thewalrus/samples.py

```diff
@@ -87,83 +89,159 @@
 # ===============================================================================================
 
 # pylint: disable=too-many-branches
-def generate_hafnian_sample(
-    cov, mean=None, hbar=2, cutoff=6, max_photons=30, approx=False, approx_samples=1e5
-):  # pylint: disable=too-many-branches
+def decompose_cov(cov):
+    r"""Decompose the convariance matrix using the Williamson decomposition method.
+
+    Args:
+        cov (array): a :math:`2N\times 2N` ``np.float64`` covariance matrix
```

Comment:
Is this statement meaning that if the matrix is not of type `np.float64` then this routine will fail? If so then a type check and perhaps a cast to the correct dtype should be added.